internal/clients/http/http.go

```diff
@@ -75,19 +82,26 @@ func (c *Client) Post(url string, headers map[string]string, body []byte) error
 
 	resp, err := c.client.Do(req)
 	if err != nil {
-		return fmt.Errorf("cannot post to URL: %w", err)
+		return nil, fmt.Errorf("cannot post to URL: %w", err)
 	}
 	defer resp.Body.Close()
 
 	data, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return fmt.Errorf("cannot read response body: %w", err)
+		return nil, fmt.Errorf("cannot read response body: %w", err)
 	}
 	log.Debugf("received HTTP %v: %v", resp.Status, strings.TrimSpace(string(data)))
 
+	apiResponse := &yggdrasil.APIresponse{
+		Code:   resp.StatusCode,
+		Body:   data,
+		Method: http.MethodGet,
+		URL:    url,
+	}
+
 	if resp.StatusCode >= 400 {
-		return &yggdrasil.APIResponseError{Code: resp.StatusCode, Body: strings.TrimSpace(string(data))}
+		return nil, apiResponse
```

Comment:
should this be `apiResponse, nil` instead?

Reply:
APIResponse implements the error() interface.